weblate/trans/views/files.py

```diff
@@ -86,18 +86,26 @@ def download_component(request, project, component):
 
 def download_project(request, project):
     obj = get_project(request, project)
-    obj.commit_pending("download", None)
+    components = obj.component_set.filter_access(request.user)
+    if not components:
+        raise PermissionDenied()
+    for component in components:
+        component.commit_pending("download", None)
```

Comment:
I'd probably stay with project wide commit as that one is way more effective in case of project with many linked components. The additional commit on not accessible components should do no harm.